build_system/clerk_driver.ml

```diff
@@ -506,44 +504,57 @@ let base_bindings catala_exe catala_flags build_dir include_dirs test_flags =
 let[@ocamlformat "disable"] static_base_rules =
   let open Var in
   let color = Message.has_color stdout in
+  let shellout l = Format.sprintf "$$(%s)" (String.concat " " l) in
   [
     Nj.rule "copy"
       ~command:["cp"; "-f"; !input; !output]
       ~description:["<copy>"; !input];
 
     Nj.rule "catala-ocaml"
-      ~command:[!catala_exe; "ocaml"; !catala_flags; !catala_flags_ocaml; !input; "-o"; !output]
+      ~command:[!catala_exe; "ocaml"; !catala_flags; !catala_flags_ocaml;
+                !input; "-o"; !output]
       ~description:["<catala>"; "ocaml"; "⇒"; !output];
 
+    Nj.rule "ocaml-object"
```

Comment:
This remains fairly constrained, at least because we have full control on the .ml sources. But you should see the Makefile rules in catala-examples :grimacing: : at that point we are indeed building full-fledged OCaml libraries that we install to the prefix... At least we can still use `ocamlfind`

The thing is, dune is really uncooperative with other build systems, so I am not sure if we have a better choice. We could consider creating a mock OCaml project in a sandbox, running dune on that and extracting the results, but I am not sure that would be simpler to handle and it would be infinitely more complex in the processing that is involved.

As long as we control the ml files, the inter-dependencies and the target libraries we want, while certainly not pleasant, working through the build through manual use of the OCaml toolchain remains doable.